### PR TITLE
refactor: edit useIsMobile hook to use useMediaQuery from mui

### PR DESCRIPTION
**Pull request by leleOFA**
chore: install css-mediaquery dev dependency with @types for testing purpose

## Short description
with the changes made it occurs that if the application is in mobile mode based on a mui hook and no longer based on the window size

## List of changes proposed in this pull request
- I entered the useMediaQuery hook referring to the control dimension with breakpoints
- I inserted in the tests a method of the css-mediaquery dependency which controls the size of the screen

## How to test
to test the refactoring change width of window initially over 1200px for desktop mode while for the mobile mode set width under 1200px

### CODE REVIEW

**Comment by clombardi** [comment]
I'm looking this PR.

**Review by clombardi — CHANGES_REQUESTED** [review]
The changes in both functionality and tests are in general OK, there are just some details to be checked.

File: packages/pn-personafisica-login/src/pages/login/__tests__/Login.test.tsx [hunk 1]
```diff
@@ -1,9 +1,10 @@
 import { render, screen, fireEvent } from '@testing-library/react';
-import { BrowserRouter } from "react-router-dom";
+import { BrowserRouter } from 'react-router-dom';
 
 import Login from '../Login';
 import { ENV } from '../../../utils/env';
 import '../../../locales/i18n';
+import React from 'react';
```
[the remark this answers is not in the document]
Comment (leleOFA) [reply packages/pn-personafisica-login/src/pages/login/__tests__/Login.test.tsx:7]: I placed them at the top

File: packages/pn-personafisica-webapp/src/component/Contacts/__test__/DigitalContactElem.test.tsx [hunk 2]
```diff
@@ -121,26 +127,28 @@ describe('DigitalContactElem Component', () => {
...
+    userEvent.click(dialogButtons![0]);
     await waitFor(() => {
       expect(dialog).not.toBeInTheDocument();
     });
   });
 
   it('closes remove modal (clicks on confirm)', async () => {
     const buttons = result?.container.querySelectorAll('button');
-    fireEvent.click(buttons![1]);
+    userEvent.click(buttons![1]);
```
[the remark this answers is not in the document]
Comment (leleOFA) [reply packages/pn-personafisica-webapp/src/component/Contacts/__test__/DigitalContactElem.test.tsx:142]: the docs that you see is for user-event@14, but we use user-event@13.5 and the setup() method is not present in this version

**Review by clombardi — APPROVED** [review]
OK, ready to merge.

**Pull request merged**